src/main/java/mate/academy/ishop/dao/hibernate/ItemDaoHibernateImpl.java

```diff
@@ -56,16 +68,50 @@ public Item update(Item item) {
             }
             LOGGER.error("Can't update item.", e);
             throw new RuntimeException();
+        } finally {
+            if (session != null) {
+                session.close();
+            }
         }
         return item;
     }
 
     @Override
     public void delete(Long id) {
+        Transaction transaction = null;
+        Session session = null;
+        try{
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Item item = new Item();
+            item.setIdItem(id);
+            session.delete(item);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            LOGGER.error("Can't delete item", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Item> getAll() {
-        return null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            return session.createCriteria(Item.class).list();
+        } catch (Exception e){
+            LOGGER.error("Can't create Criteria(Item.class)", e);
+            throw new RuntimeException();
```

Comment:
```suggestion
            throw new RuntimeException(e);
```
@Sasha192  По перше ти викидуєш пусту помилку. Це не інформативно. По друге ти викидуєш загальну RuntimeException, а не якусь більш специфічну, що також не інформативно. По третє: в аналогічних методах в своєму рішенні ти повертаєш пустий список, чому вирішив тут викинути Exception?

Reply:
краще просто повернути пустий ліст